### ISSUE TITLE
Slow rule consumes over 10GB of memory - react-css-injection

### ISSUE THREAD

**Issue by radekk** [returntocorp/sgrep-rules#1114]
**Describe the bug**

When running a specific rule from the typescript set it's not returning results. Instead, it runs continuously, and throws an error after the whole free memory available for this process is being consumed. It took over 10GB of memory before exiting.

Semgrep version: `0.40.0`
Rule: `./semgrep-rules/typescript/react/security/audit/react-css-injection.yaml`
Rule ID: `semgrep-rules.typescript.react.security.audit.react-css-injection`

**To Reproduce**

Create a new file (`slow_file.ts`):

```typescript
export function foobar(
  number: number,
  params?: Record<string, any>
) {
  return Array(number)
    .fill(true)
    .map(() => {
      return bar(params);
    });
}
```

As the next step run semgrep  command:

```shell
semgrep -f ./semgrep-rules/typescript/react/security/audit/react-css-injection.yaml slow_file.ts 
```

Output:

```shell
running 1 rules...
an internal error occured while invoking semgrep-core while running rule 'semgrep-rules.typescript.react.security.audit.react-css-injection'. Consider skipping this rule and reporting this issue.
	unknown exception: Stack overflow
An error occurred while invoking the semgrep engine; please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
```

**Expected behavior**

It should return back results in a reasonable time for such operation.

**Additional Context**

I found this issue when I tried to run the `typescript` rule set on the typescript project. I was able to track it down to the specific rule and then find a code that was not being processed properly. Let me know if you need any other details.

**Comment by radekk** [comment]
The same issue is happening when parsing mentioned `slow_file.ts` with rules:
- react-jwt-decoded-property
- react-insecure-request
- react-props-in-state

After removing all those rules I can scan using `typescript` rule set. Other rules in this set seems to be working fine.

**Comment by aryx** [comment]
I can't reproduce on develop. I think it's related to https://github.com/returntocorp/semgrep/pull/2584

**Comment by aryx** [comment]
This should be fixed in 0.41 that we will release today.
Thx for the bug report!
